packages/language-server/src/server.ts

```diff
@@ -54,7 +78,8 @@ connection.onCompletion(
       const documentUri = textDocumentPosition.textDocument.uri;
       const document = documents.get(documentUri);
       if (document) {
-        return getCompletionItems(model, textDocumentPosition, document);
+        updateDocumentSettings(document.uri);
```

Comment:
Why do we need to `updateDocumentSettings` on every completion request?

Reply:
There are 2 flows for settings, depending on the client capabilities:
1. The client doesn't support workspace/document-level settings (`workspace/configuration` request). In this case we use global settings (which arrive in `onDidChangeConfiguration`) and don't try to fetch the workspace/document settings.
2. The client supports `workspace/configuration` request. In this case we ask for the document's settings when we need them if we don't have them, and clear all the document settings when the configuration changes (`onDidChangeConfiguration`).
Currently we need the settings only for the code completion, so this is where we do the call . It isn't done on every request, only if we don't have the document settings (this logic is inside `updateDocumentSettings`).

This logic is based on the [language server extension guide in vscode documentation](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide).

Reply:
Would it be clearer if I renamed `updateDocumentSettings` to `ensureDocumentSettings`?

Reply:
Renamed and added a comment